clang/lib/StaticAnalyzer/Checkers/MallocChecker.cpp

```diff
@@ -103,14 +103,49 @@ using namespace std::placeholders;
 namespace {
 
 // Used to check correspondence between allocators and deallocators.
-enum AllocationFamily {
+enum AllocationFamilyKind {
   AF_None,
   AF_Malloc,
   AF_CXXNew,
   AF_CXXNewArray,
   AF_IfNameIndex,
   AF_Alloca,
-  AF_InnerBuffer
+  AF_InnerBuffer,
+  AF_Custom,
+};
+
+struct AllocationFamily {
+  AllocationFamilyKind Kind;
+  std::optional<StringRef> CustomName;
+
+  explicit AllocationFamily(AllocationFamilyKind kind,
+                            std::optional<StringRef> name = std::nullopt)
+      : Kind(kind), CustomName(name) {
+    assert(kind != AF_Custom || name != std::nullopt);
```

Comment:
I'd prefer to have a message here to be able to immediately tell what this assert is about when fires.
```suggestion
    assert((kind != AF_Custom || name.has_value()) && "Custom family must specify also the name");
```